SSINS/sky_subtract.py

```diff
@@ -173,6 +193,32 @@ def diff(self):
         super().set_lsts_from_time_array()
         self.data_array = np.ma.masked_array(self.data_array)
 
+    def diff_freq(self):
+
+        """
+        Differences the visibilities in freq. Does so independently for each time.
+        The flags are propagated by taking the boolean OR of the entries that correspond
+        to the visibilities that are differenced from one another. Other metadata
+        attributes are also adjusted so that the resulting SS object passes
+        UVData.check()
+        """
+
+        if self.blt_order != 'baseline':
+            warnings.warn("Reordering data array to baseline order to perform differencing.")
+            self.reorder_blts(order='baseline')
+
+        diff_dat = np.diff(self.data_array, axis=2) #axis 2: nfreqs (see uvdata object)
+        self.data_array = diff_dat
+
+        print(self.data_array.shape)
+        """The frequency-differenced visibilities. Complex array of shape (Nblts, Nspws, Nfreqs, Npols)."""
```

Comment:
This may or may not show up in the docs depending on where this code lives relative to `diff`. The general docstring for the data_array should probably just say `The visibilities, which may be differenced in time or frequency depending on settings...`

Reply:
https://github.com/mwilensky768/SSINS/pull/91/commits/9402d567c6879e7a3d3d06fa8935bfe836ab466b resolved this.